indico/modules/events/contributions/__init__.py

```diff
@@ -104,10 +104,12 @@ def _extend_event_menu(sender, **kwargs):
     from indico.modules.events.layout.util import MenuEntryData
 
     def _visible_my_contributions(event):
-        return session.user and has_contributions_with_user_as_submitter(event, session.user)
+        published = contribution_settings.get(event, 'published')
+        return session.user and has_contributions_with_user_as_submitter(event, session.user) and published
```

Comment:
Maybe you could put `and published` before `and has_contributions`. This way we avoid querying the DB in case `published` is `False`.